conan/tools/files/files.py

```diff
@@ -0,0 +1,152 @@
+             md5=md5, sha1=sha1, sha256=sha256)
+    unzip(filename, destination=destination, keep_permissions=keep_permissions, pattern=pattern,
+          output=output, strip_root=strip_root)
+    os.unlink(filename)
+
+
+def ftp_download(conanfile, ip, filename, login='', password=''):
+    import ftplib
+    try:
+        ftp = ftplib.FTP(ip)
+        ftp.login(login, password)
+        filepath, filename = os.path.split(filename)
+        if filepath:
+            ftp.cwd(filepath)
+        with open(filename, 'wb') as f:
+            ftp.retrbinary('RETR ' + filename, f.write)
+    except Exception as e:
+        try:
+            os.unlink(filename)
+        except OSError:
+            pass
+        raise ConanException("Error in FTP download from %s\n%s" % (ip, str(e)))
+    finally:
+        try:
+            ftp.quit()
+        except Exception:
+            pass
+
+
+def download(conanfile, url, filename, verify=True, out=None, retry=None, retry_wait=None, overwrite=False,
+             auth=None, headers=None, requester=None, md5='', sha1='', sha256=''):
+    """Retrieves a file from a given URL into a file with a given filename.
+       It uses certificates from a list of known verifiers for https downloads,
+       but this can be optionally disabled.
+
+    :param url: URL to download. It can be a list, which only the first one will be downloaded, and
+                the follow URLs will be used as mirror in case of download error.
+    :param filename: Name of the file to be created in the local storage
+    :param verify: When False, disables https certificate validation
+    :param out: An object with a write() method can be passed to get the output. stdout will use if
+                not specified
+    :param retry: Number of retries in case of failure. Default is overriden by general.retry in the
+                  conan.conf file or an env variable CONAN_RETRY
+    :param retry_wait: Seconds to wait between download attempts. Default is overriden by
+                       general.retry_wait in the conan.conf file or an env variable CONAN_RETRY_WAIT
+    :param overwrite: When True, Conan will overwrite the destination file if exists. Otherwise it
+                      will raise an exception
+    :param auth: A tuple of user and password to use HTTPBasic authentication
+    :param headers: A dictionary with additional headers
+    :param requester: HTTP requests instance
+    :param md5: MD5 hash code to check the downloaded file
+    :param sha1: SHA-1 hash code to check the downloaded file
+    :param sha256: SHA-256 hash code to check the downloaded file
+    :return: None
+    """
+    # TODO: Add all parameters to the new conf
+    out = conanfile.output
+    requester = conanfile._conan_requester
+    config = conanfile.conf
+
+    # It might be possible that users provide their own requester
+    retry = retry if retry is not None else int(config["tools.files.download"].retry)
```

Comment:
To be discussed: if recipe defines retry=1, that is hardcoded. Trying to configure it downstream in [conf] will have no effect, I am not sure about this. 

We are good as-is now, but we want to review this later.